Buddy/View/UserProfile.cs

```diff
@@ -0,0 +1,140 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.IO;
+using System.Linq;
+using Buddy.Data;
+using Buddy.Data.Const;
+using Buddy.Data.Entity;
+using Buddy.Logic;
+using Buddy.Repository;
+using AppContext = Buddy.Data.AppContext;
+
+namespace Buddy.View
+{
+    public partial class UserProfile : BaseMetro
+    {
+        private readonly IUserRepository _userRepository;
+        private readonly IRepository<Post> _postRepository;
+        private readonly IPostRecordRepository _postRecordRepository;
+        private readonly PostRecordManager _postRecordManager;
+        private readonly PostManager _postManager;
+
+        private List<Post> _posts;
+
+        public UserProfile(IRepository<Post> postRepository,
+            PostManager postManager,
+            IUserRepository userRepository,
+            IPostRecordRepository postRecordRepository,
+            PostRecordManager postRecordManager)
+        {
+            _postRepository = postRepository;
+            _postManager = postManager;
+            _userRepository = userRepository;
+            _postRecordRepository = postRecordRepository;
+            _postRecordManager = postRecordManager;
+            InitializeComponent();
+        }
+
+        private void UserProfile_Load(object sender, EventArgs e)
+        {
+            labelUsername.Text = $@"{CurrentUser.UserName}";
+
+            if (CurrentUser.ProfilePicture != null) {
+                //byte[] => picture
+                using (MemoryStream mStream = new MemoryStream(CurrentUser.ProfilePicture))
+                {
+                    pictureboxProfilePic.Image = Image.FromStream(mStream);
+                }
+            }
+            else if (CurrentUser.ProfilePicture == null) {
+
+                 pictureboxProfilePic.Image = Image.FromFile(@"..\..\Resources\ProfileDefault.jpg");
+            }
+        }
+
+        private void metrolinkPosts_Click(object sender, EventArgs e)
+        {
+            this.userInfo1.Hide();
+            this.metrogrid.DataSource = null;
+            var context = new AppContext();
+            var posts = context.Posts.ToList();
```

Comment:
```suggestion
var posts = _postRepository.GetAll();
```
using context directly is again not needed